### PR TITLE
Encapsulate field dialog

### PR DESCRIPTION
**Pull request by Hosch250**

### CODE REVIEW

File: RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public override void Fix()
         {
             var vbe = _target.Project.VBE;
 
-            using (var view = new EncapsulateFieldDialog(_state, _indenter))
+            using (var view = new EncapsulateFieldDialog(new EncapsulateFieldViewModel(_state, _indenter)))
```
Comment (retailcoder) [inline RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs]: how do you test the `Fix()` method with a hard dependency on `EncapsulateFieldDialog`?

Comment (Hosch250) [inline RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs]: You don't, because it just calls the refactoring anyway.

Comment (retailcoder) [inline RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs]: So, you can't test that `Fix()` calls the refactoring; should the refactoring is considered an implementation detail?

Comment (Hosch250) [inline RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs]: I'm perfectly fine with doing an injection.

Comment (retailcoder) [inline RetailCoder.VBE/Inspections/QuickFixes/EncapsulateFieldQuickFix.cs]: I'm thinking we should inject everything, up to the inspection's constructor - and eventually give ourselves some `IInspectionResultFactory<TInspectionResult>`; when we do, we won't have to be bothered with UI couplings, and the dependencies will move from the inspection to the factory's constructor... and if we manage to wire up a Ninject auto-factory, we won't even need to think about them.

File: RetailCoder.VBE/Refactorings/EncapsulateField/EncapsulateFieldPresenter.cs [hunk 2]
```diff
@@ -46,50 +47,51 @@ public EncapsulateFieldModel Show()
```
Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/EncapsulateField/EncapsulateFieldPresenter.cs]: I think starting a line 38 we have a new method - there's something that doesn't feel right about that `if..else` tree here. Whether the viewmodel `CanHaveLet` or `CanHaveSet` would be much clearer if it were one statement - perhaps with some functions for proper abstraction, but I'm pretty sure I don't like that `if..else` tree.

File: RetailCoder.VBE/Refactorings/EncapsulateField/EncapsulateFieldPresenterFactory.cs [hunk 3]
```diff
@@ -7,10 +8,10 @@ namespace Rubberduck.Refactorings.EncapsulateField
     public class EncapsulateFieldPresenterFactory : IRefactoringPresenterFactory<EncapsulateFieldPresenter>
```
Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/EncapsulateField/EncapsulateFieldPresenterFactory.cs]: is it ninject injecting a `IRefactoringPresenterFactory<EncapsulateFieldPresenter>`? If we find a way to bind that constructed generic interface `ToFactory`, then the concrete implementations can just go away.

Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/EncapsulateField/EncapsulateFieldPresenterFactory.cs]: Not a mergestopper, just keeping a note that we have quite a few concrete factory implementations - that would be part of the `RubberduckModule` technical debt IMO.

**Pull request merged**